app/containers/ExplorePage.tsx

```diff
@@ -51,45 +52,110 @@ function chooseFile(
 export default function ExplorePage() {
   const { t } = useTranslation();
   const [filePath, setFilePath] = useState<string>();
+  const [filters, setFilters] = useState<object>({
+    activeAnimals: [],
+    activeCameras: [],
+    activeStations: [],
+    activeChecks: []
+  });
   const [data, setData] = useState<undefined | ObservationsData>();
+  const handleFilters = val => {
+    setFilters({ ...filters, ...val });
+  };
+
+  const filterCondition = (needle, haystack) => {
+    if (haystack.length === 0) return true;
+    return haystack.map(entry => entry.value).includes(needle);
+  };
+
+  const getFilteredData = options => {
+    let filtered = options.observations;
+
+    if (typeof filters !== 'undefined') {
+      filtered = filtered.filter(entry => {
+        const validEntry =
+          filterCondition(entry.pred_1, filters.activeAnimals) &&
+          filterCondition(entry.camera, filters.activeCameras) &&
+          filterCondition(entry.station, filters.activeStations) &&
+          filterCondition(entry.check, filters.activeChecks);
+        return validEntry;
+      });
+    }
+    return { observations: filtered };
+  };
+
+  const MainPanel: React.SFC = () => (
+    <div style={{ display: 'flex' }}>
+      <div
+        style={{
+          flex: '1',
+          width: '55%',
+          paddingBottom: '20px',
+          marginRight: '10px'
+        }}
+      >
+        <Card style={{ height: '100%' }} interactive elevation={Elevation.TWO}>
+          <AnimalsPlot data={getFilteredData(data)} />
+        </Card>
+      </div>
+      <div
+        style={{
+          flex: '1',
+          width: '45%',
+          paddingBottom: '20px',
+          marginLeft: '10px'
+        }}
+      >
+        <Card style={{ height: '100%' }} interactive elevation={Elevation.TWO}>
+          <Map data={getFilteredData(data)} />
+        </Card>
+      </div>
+    </div>
+  );
+  const TablePanel: React.SFC = () => (
+    <ObservationsTable style={{ width: '100%' }} data={getFilteredData(data)} />
+  );
 
   const contents =
     data !== undefined ? (
       <>
-        <h1>{t('Explore')}</h1>
-        <Button
-          text={t('Back')}
-          icon="arrow-left"
-          onClick={() => setData(undefined)}
-          style={{ marginBottom: '10px', backgroundColor: '#fff' }}
-        />
-        ,
-        {/* <div style={{ flex: 1, paddingBottom: '20px' }}>
-          <Card interactive elevation={Elevation.TWO}>
-            <h2 style={{ marginTop: 0 }}>{t('Filters')}</h2>
-          </Card>
-        </div> */}
-        ,
-        <div style={{ flex: 1, width: '100%', paddingBottom: '20px' }}>
-          <Card interactive elevation={Elevation.TWO}>
-            <AnimalsPlot data={data} />
-          </Card>
+        <div>
+          <h1>{t('Explore')}</h1>
```

Comment:
maybe we should remove these titles completely? aren't they a repetition from of what you see in the menu on the left? or do you think they are useful

Reply:
Because we have a very shallow structure it might make sense to remove them. I will try it to see the effect

Reply:
Removed in the current version, it looks nice!